src/main/java/seedu/address/logic/commands/RemarkCommand.java

```diff
@@ -0,0 +1,107 @@
+        + "by the index number used in the last person listing. "
+        + "Existing remark will be overwritten by the input.\n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + PREFIX_REMARK + "[REMARK]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_REMARK + "Likes to swim.";
+
+    private static final String MESSAGE_ADD_REMARK_SUCCESS = "Added remark to Person: %1$s";
+    private static final String MESSAGE_DELETE_REMARK_SUCESS = "Removed remark from Person: %1$s";
+    private static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+
+    private final Index index;
+    private final Remark remark;
+
+    /**
+    * @param index of the person in the filtered person list to edit the remark
+    * @param remark of the person
+    */
+    public RemarkCommand(Index index, Remark remark) {
+        requireNonNull(index);
+        requireNonNull(remark);
+
+        this.index = index;
+        this.remark = remark;
+    }
+
+    @Override
+    protected CommandResult executeUndoableCommand() throws CommandException {
+
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        ReadOnlyPerson personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                personToEdit.getAddress(), remark, personToEdit.getTags());
+
+        try {
+            model.updatePerson(personToEdit, editedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        } catch (PersonNotFoundException pnfe) {
+            throw new AssertionError("The target person cannot be missing");
+        }
+        //model.updateFilteredListToShowAll();
+
+        return new CommandResult(generateSuccessMessage(editedPerson));
+    }
+
+    /**
+     * Returns sucess message depends on different scenario such as adding or deleting remarks
+     */
+    private String generateSuccessMessage(ReadOnlyPerson personToEdit) {
+        if (!remark.value.isEmpty()) {
+            return String.format(MESSAGE_ADD_REMARK_SUCCESS, personToEdit);
+        } else {
+            return String.format(MESSAGE_DELETE_REMARK_SUCESS, personToEdit);
+        }
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        //short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        //instanceof handles null
+        if (!(other instanceof RemarkCommand)) {
+            return false;
+        }
+
+        //state check
+        RemarkCommand e = (RemarkCommand) other;
+        return index.equals(e.index) && remark.equals(e.remark);
+    }
+    @Override
```

Comment:
Add a blank line above the `@Override` annotation.

Reply:
Noted. Thanks, added a blank line.